tissuumaps/static/js/utils/regionUtils.js

```diff
@@ -1798,24 +1798,44 @@ regionUtils.JSONToRegions= function(filepath){
         if (path != null) {
             filepath = path + "/" + filepath;
         }
-        fetch(filepath)
-        .then((response) => {
-            return response.json();
-        })
-        .then((regionsobj) => {
-            regionUtils.JSONValToRegions(regionsobj);
-        });
+        if (filepath.includes(".pbf")) {
+            // Load GeoJSON stored in Geobuf format (https://github.com/mapbox/geobuf)
+            fetch(filepath)
+            .then((response) => {
+                return response.arrayBuffer();
+            })
+            .then((regionsobj) => {
+                const data = new Pbf(regionsobj);
+                regionUtils.JSONValToRegions(geobuf.decode(data));
+            });
```

Comment:
Why not combine the two `then`s?

Reply:
Ah, didn't realize that `arrayBuffer` is `async`, too. My mistake!

Reply:
Yes, it only returns a promise, so both were needed.